crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -36,18 +36,21 @@ pub(crate) struct RecursorDnsHandle {
     name_server_cache: Arc<Mutex<NameServerCache<TokioRuntimeProvider>>>,
     record_cache: DnsLru,
     recursion_limit: u8,
+    ns_recursion_limit: u8,
```

Comment:
Would prefer using `Option` instead of using `0` as a sentinel. (Could technically use `Option<NonZero<u8>>` but not that's not supported by our MSRV -- `NonZeroU8` is -- and might be overkill?)

Reply:
I changed this, and also changed the standard query recursion limit to follow the same form.